src/storage/hummock_sdk/src/filter_key_extractor.rs

```diff
@@ -152,29 +174,56 @@ impl FilterKeyExtractor for SchemaFilterKeyExtractor {
         let (_vnode_prefix, pk) = key.split_at(VirtualNode::SIZE);
 
         // if the key with table_id deserializer fail from schema, that should panic here for early
-        // detection
-        let pk_prefix_len = self
-            .deserializer
-            .deserialize_prefix_len_with_column_indices(pk, 0..self.read_pattern_prefix_column)
-            .unwrap();
-
-        let prefix_len = TABLE_PREFIX_LEN + VirtualNode::SIZE + pk_prefix_len;
-        &full_key[0..prefix_len]
+        // detection.
+        match self.distribution_key_indices_pair_in_pk.is_some()
```

Comment:
Using `if let Some((.., ..)) = self.distribution_key_indices_pair_in_pk` seems more natural to me.